node-deb

```diff
@@ -683,25 +683,39 @@ find "$@" -type f -print0 | {
   done
 }
 
-if [ -d "$source_dir/node_modules" ]; then
-  npm ls --parseable --prod | sed -e "s/$(escape "$source_dir")//g" | grep -Ev '^$' | sed -e 's:/node_modules/::g' | {
-    while IFS="\n" read -r -d '' dir; do
-      log_debug "Copying dir: $dir"
-      cp -rf "$source_dir/node_modules/$dir" "$deb_dir/usr/share/$package_name/app/node_modules/"
-    done
-  }
-fi
+app_dir="$deb_dir/usr/share/$package_name/app"
 
-if ! [ -f "$deb_dir/usr/share/$package_name/app/package.json" ]; then
+if ! [ -f "$app_dir/package.json" ]; then
   log_info "Including 'package.json' in the Debian package."
   cp -pf './package.json' "$deb_dir/usr/share/$package_name/app/"
 fi
 
-if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$deb_dir/usr/share/$package_name/app/npm-shrinkwrap.json" ]; then
+if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$app_dir/npm-shrinkwrap.json" ]; then
   log_info "Including 'npm-shrinkwrap.json' in the Debian package."
-  cp -pf './npm-shrinkwrap.json' "$deb_dir/usr/share/$package_name/app/"
+  cp -pf './npm-shrinkwrap.json' "$app_dir"
+fi
+
+if [ -f './yarn.lock' ] && ! [ -f "$app_dir/yarn.lock" ]; then
+  log_info "Including 'yarn.lock' in the Debian package."
+  cp -pf './yarn.lock' "$app_dir"
 fi
 
+current_dir="$(pwd)"
+cd "$app_dir"
+if [ -f "./yarn.lock" ]; then
+  log_info "Building node_modules dir with yarn"
+  yarn install --no-progress --production 2>&1 > /dev/null | sed "s/^/YARN: /"
```

Comment:
nope. the statement is correct this way. it basically pipes only stderr to sed, but in a way that sed keeps prefixing output with yarn/npm. i didn’t come up with it :). see http://stackoverflow.com/a/2342841